iree/compiler/Dialect/VM/Conversion/VMToEmitC/ConvertVMToEmitC.cpp

```diff
@@ -0,0 +1,98 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "iree/compiler/Dialect/VM/Conversion/VMToEmitC/ConvertVMToEmitC.h"
+
+#include "emitc/Dialect/EmitC/EmitCDialect.h"
+#include "iree/compiler/Dialect/VM/IR/VMOps.h"
+#include "mlir/IR/Matchers.h"
+#include "mlir/IR/Module.h"
+#include "mlir/Transforms/DialectConversion.h"
+
+namespace mlir {
+namespace iree_compiler {
+
+namespace {
+
+// Taken over from StandartToVM.
+// We need to replace the Op depending on the operand.
+// We could start with a conversion for IREE::VM::AddI32Op
+template <typename SrcOpTy, typename DstOpTy>
+class BinaryArithmeticOpConversion : public OpConversionPattern<SrcOpTy> {
+  using OpConversionPattern<SrcOpTy>::OpConversionPattern;
+
+ public:
+  BinaryArithmeticOpConversion(MLIRContext *context, StringRef funcName)
+      : OpConversionPattern<SrcOpTy>(context), funcName(funcName) {}
+
+ private:
+  LogicalResult matchAndRewrite(
+      SrcOpTy srcOp, ArrayRef<Value> operands,
+      ConversionPatternRewriter &rewriter) const override {
+    typename SrcOpTy::Adaptor srcAdapter(operands);
+
+    MLIRContext *ctx = srcOp.getContext();
+
+    // name of the function to call
+    StringAttr callee_ = StringAttr::get(funcName, ctx);
+
+    // attributes of the function call; references only the variable operands
+    ArrayAttr args_ = ArrayAttr::get({IntegerAttr::get(IndexType::get(ctx), 0),
+                                      IntegerAttr::get(IndexType::get(ctx), 1)},
+                                     ctx);
```

Comment:
These are exposed in the rewriter, do avoid needing an explicit context. I'm not really sure it's worth it for those methods to exist, but they do, so might as well use them :grin:

```suggestion
    // name of the function to call
    StringAttr callee_ = rewriter.getStringAttr(funcName);

    // attributes of the function call; references only the variable operands
    ArrayAttr args_ = rewriter.getArrayAttr(
        {rewriter.getIntegerAttr(rewriter.getIndexType(), 0),
         rewriter.getIntegerAttr(rewriter.getIndexType(), 1)});
```

The one for IntegerAttr is pretty useless IMO, so maybe leave that one as is

```suggestion
    // name of the function to call
    StringAttr callee_ = rewriter.getStringAttr(funcName);

    // attributes of the function call; references only the variable operands
    ArrayAttr args_ =
        rewriter.getArrayAttr({IntegerAttr::get(rewriter.getIndexType(), 0),
                               IntegerAttr::get(rewriter.getIndexType(), 1)});
```

Reply:
Ah thanks, I wasn't aware of that. :) I've gone with the second option.

Reply:
Seems Simon has fixed this, therefore marking as resolved.